pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -286,9 +291,21 @@ func deploymentOnPVC(c *Cluster, osd OSDInfo, pvcName string, config *provisionC
 		return nil, errors.Wrapf(err, "failed to generate deployment for %s", osdLongName)
 	}
 
+	err = prepareForDeployment(c, osdProps, osd)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to prepare for deployment %s", osdLongName)
+	}
+
 	return d, nil
 }
 
+func prepareForDeployment(c *Cluster, osdProps osdProperties, osd OSDInfo) error {
+	if osdProps.storeConfig.PrimaryAffinity != "" {
+		return cephclient.OsdPrimaryAffinity(c.context, c.clusterInfo, osd.ID, osdProps.storeConfig.PrimaryAffinity)
```

Comment:
I'm not excited about setting this property with every reconcile, so would prefer if we can set it with a command line param.